testutil/cmdtest/system.go

```diff
@@ -0,0 +1,120 @@
+// Package cmdtest contains a framework for testing cobra Commands within Go unit tests.
+package cmdtest
+
+import (
+	"bytes"
+	"context"
+	"io"
+
+	"github.com/spf13/cobra"
+)
+
+// System is a system under test.
+type System struct {
```

Comment:
This is cool; however, I find the term `System` a bit ambiguous.
I don't have a better term 😅, but it is not immediately clear that it is for testing cobra commands.